chain_test.go

```diff
@@ -0,0 +1,99 @@
+//go:build linux
+// +build linux
+
+package netlink
+
+import (
+	"testing"
+
+	"golang.org/x/sys/unix"
+)
+
+func TestChainAddDel(t *testing.T) {
+	tearDown := setUpNetlinkTest(t)
+	defer tearDown()
+	if err := LinkAdd(&Ifb{LinkAttrs{Name: "foo"}}); err != nil {
+		t.Fatal(err)
+	}
+	if err := LinkAdd(&Ifb{LinkAttrs{Name: "bar"}}); err != nil {
+		t.Fatal(err)
+	}
+	link, err := LinkByName("foo")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := LinkSetUp(link); err != nil {
+		t.Fatal(err)
+	}
+	redir, err := LinkByName("bar")
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := LinkSetUp(redir); err != nil {
+		t.Fatal(err)
+	}
+	qdisc := &Ingress{
+		QdiscAttrs: QdiscAttrs{
+			LinkIndex: link.Attrs().Index,
+			Handle:    MakeHandle(0xffff, 0),
+			Parent:    HANDLE_INGRESS,
+		},
+	}
+	if err := QdiscAdd(qdisc); err != nil {
+		t.Fatal(err)
+	}
+	qdiscs, err := SafeQdiscList(link)
```

Comment:
i think we can assume `QdiscAdd` works as expected (as its functionality is covered by Qdisc related tests.
so IMO you can remove L45-55 WDYT ?

Reply:
I prefer to keep these lines for consistency and to make TestChainAddDel more explicit in case of failure

Reply:
even more: such tests helped me to find an issue with filter chain implementation